src/orient/rational-calc.orient

```diff
@@ -0,0 +1,704 @@
+    describe(post_snark_constraints, "Constraints for a PoSt")
+    describe(post_snark_partitions, "Number of partitions of a PoSt proof")
+    describe(post_snark_size, "Size of a PoSt proof", bytes)
+    describe(post_snark_size_kib, "Size of a Post proof", kib)
+    describe(post_snark_partition_constraints, "Number of constraints for a PoSt partition")
+    describe(post_snark_partition_time, "Time to generate a single PoSt SNARK partition", secs)
+    describe(commr_column_size, "Number of nodes in the CommR leaf")
+
+    // TODO: make sure VectorR/etc are handled correctly
+    inclusion_proof = merkle_tree_hash_constraints * commr_tree_depth 
+    post_inclusion_proofs = inclusion_proof * post_challenges
+    ticket_proofs = ticket_constraints * commr_column_size * post_challenges
+    // TODO: handle different constructions
+    post_snark_constraints = ticket_proofs + post_inclusion_proofs
+    post_snark_partitions = post_snark_constraints / post_snark_partition_constraints
+    post_snark_size = post_snark_partitions * snark_size
+    post_snark_size_kib = post_snark_size / b_in_kib
+    post_snark_time = snark_constraint_time * post_snark_constraints
+    post_snark_partition_time = snark_constraint_time * post_snark_constraints
+    // Parallelization calculation: it must take at least the time for a single snark
+    // the remaining can be done in parallel
+    // Similar in spirit to:
+    // min(porep_snark_partition_time, porep_snark_time / snarrk_parallelization)
+    post_snark_time_parallel = post_snark_partition_time + (post_snark_time - post_snark_partition_time) / (rig_snark_parallelization)
+
+ProofOfReplication:
+  Security:
+    describe(porep_challenges, "Number of challenges in a PoRep proof")
+    describe(porep_lambda, "Bits of security of a PoRep proof")
+    describe(spacegap, "Maximum difference in storage between an honest prover and a malicious prover")
+    describe(porep_soundness, "Probability of an adversary to succeed in generating a convincing PoRep proof")
+
+    porep_soundness = expt(2, -porep_lambda)
+
+  Graph:
+    describe(nodes, "Number of nodes in a sector")
+    describe(stacked_layers, "Number of stacked layers of the graph")
+
+    Sector:
+      describe(sector_size, "Size of a sector", bytes)
+      describe(sector_size_mib, "Size of a sector", mib)
+      describe(sector_size_gib, "Size of a sector", gib)
+
+      sector_size = sector_size_gib * b_in_gib
+      sector_size_mib = sector_size / b_in_mib 
+
+    Window:
+      describe(window_nodes, "Number of nodes in a window")
+      describe(window_size, "Size of a window", bytes)
+      describe(window_size_mib, "Size of a window", mib)
+      describe(window_size_gib, "Size of a window", mib)
+      describe(windows, "Number of windows in a sector")
+
+      window_size = window_size_mib * b_in_mib 
+      window_size = window_size_gib * b_in_gib 
+      windows = sector_size / window_size
```

Comment:
Is this guaranteed to be an integer? Is there a difference between a float and an integer division?